In [3]:
%pip install roboflow

from roboflow import Roboflow
rf = Roboflow(api_key="XLCwBcZbaxnQjKvVoY2a")
project = rf.workspace("viren-dhanwani").project("tennis-ball-detection")
version = project.version(6)
dataset = version.download("yolov5")



[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.
loading Roboflow workspace...
loading Roboflow project...



Extracting Dataset Version Zip to tennis-ball-detection-6 in yolov5pytorch:: 100%|██████████| 1168/1168 [00:00<00:00, 1819.16it/s]


In [4]:
import shutil
shutil.move("tennis-ball-detection-6/train",
"tennis-ball-detection-6/tennis-ball-detection-6/train",
)
shutil.move("tennis-ball-detection-6/test",
"tennis-ball-detection-6/tennis-ball-detection-6/test",
)
shutil.move("tennis-ball-detection-6/valid",
"tennis-ball-detection-6/tennis-ball-detection-6/valid",
)

'tennis-ball-detection-6/tennis-ball-detection-6/valid'

In [20]:
!yolo task=detect mode=train model=yolov5l6u.pt data={dataset.location}/data.yaml epochs=50 imgsz=640

Python(69048) MallocStackLogging: can't turn off malloc stack logging because it was not enabled.


New https://pypi.org/project/ultralytics/8.1.28 available 😃 Update with 'pip install -U ultralytics'
Ultralytics YOLOv8.1.27 🚀 Python-3.12.2 torch-2.2.1 MPS (Apple M2 Pro)
engine/trainer: task=detect, mode=train, model=yolov5l6u.pt, data=/Users/evgeny.tolmachev/Personal/TrackBall/training/tennis-ball-detection-6/data.yaml, epochs=50, time=None, patience=100, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=mps, workers=8, project=None, name=train4, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina

In [16]:
!yolo task=detect mode=predict model=yolov5l6u.pt source="../images/tennis_match.jpg"

Python(68822) MallocStackLogging: can't turn off malloc stack logging because it was not enabled.


Ultralytics YOLOv8.1.27 🚀 Python-3.12.2 torch-2.2.1 CPU (Apple M2 Pro)
YOLOv5l6u summary (fused): 393 layers, 86053520 parameters, 0 gradients

image 1/1 /Users/evgeny.tolmachev/Personal/TrackBall/training/../images/tennis_match.jpg: 768x1280 13 persons, 1 backpack, 1 sports ball, 1 chair, 734.8ms
Speed: 2.8ms preprocess, 734.8ms inference, 0.8ms postprocess per image at shape (1, 3, 768, 1280)
Results saved to /Users/evgeny.tolmachev/Personal/TrackBall/runs/detect/predict9
💡 Learn more at https://docs.ultralytics.com/modes/predict


In [18]:
!yolo task=detect mode=predict model=yolov5l6u.pt source="../images/tennis_match.jpg" device=mps

Python(68953) MallocStackLogging: can't turn off malloc stack logging because it was not enabled.


Ultralytics YOLOv8.1.27 🚀 Python-3.12.2 torch-2.2.1 MPS (Apple M2 Pro)
YOLOv5l6u summary (fused): 393 layers, 86053520 parameters, 0 gradients

image 1/1 /Users/evgeny.tolmachev/Personal/TrackBall/training/../images/tennis_match.jpg: 768x1280 13 persons, 1 backpack, 1 sports ball, 1 chair, 515.1ms
Speed: 9.5ms preprocess, 515.1ms inference, 100.9ms postprocess per image at shape (1, 3, 768, 1280)
Results saved to /Users/evgeny.tolmachev/Personal/TrackBall/runs/detect/predict11
💡 Learn more at https://docs.ultralytics.com/modes/predict


Speed: 2.9ms preprocess, 770.6ms inference, 1.1ms postprocess per image at shape (1, 3, 768, 1280)
